### PR TITLE
Feature/e UI 4749 5299

### PR DESCRIPTION
**Pull request by udaydenduluri33**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-4714
https://tools.hmcts.net/jira/browse/EUI-5299
https://tools.hmcts.net/jira/browse/EUI-4749

### Change description ###
Adding roles and access section for admin roles
Adding role allocation journey for admin users (note: can't actually allocate via API yet)
Adding error page when no roles available for service and role category specified

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x ] No
```

### CODE REVIEW

File: src/cases/components/role-access-section/role-access-section.component.html [hunk 1]
```diff
@@ -0,0 +1,26 @@
+<h2 class="govuk-heading-m">{{title}}</h2>
+<div *ngIf="!namedRoles">
+  <exui-case-roles-table [roles]="roles"
+                        [roleCategory]="roleCategory"
+                        [caseDetails]="caseDetails"
+                        [showAllocateRoleLink]="showAllocateRoleLink">
+  </exui-case-roles-table>
+</div>
+<div *ngIf="namedRoles">
+  <exui-case-roles-table [roles]="namedRoles"
```
Comment (scott-turner-hmcts) [inline src/cases/components/role-access-section/role-access-section.component.html:10]: @connorpgpmcelroy code could be refactored here to have only one <exui-case-roles-table> e.g:

`<exui-case-roles-table [roles]="namedRoles ? namedRoles : roles">...`

File: src/cases/components/roles-and-access/roles-and-access.component.spec.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import { async, ComponentFixture, TestBed } from '@angular/core/testing';
 import { RouterTestingModule } from '@angular/router/testing';
 import { ExuiCommonLibModule } from '@hmcts/rpx-xui-common-lib';
+import { RoleAccessSectionComponent } from '../role-access-section/role-access-section.component';
```
Comment (scott-turner-hmcts) [inline src/cases/components/roles-and-access/roles-and-access.component.spec.ts:4]: very minor, but could this import be grouped with the one on line 13 ?

File: src/cases/components/roles-and-access/roles-and-access.component.ts [hunk 3]
```diff
@@ -13,6 +13,8 @@ export class RolesAndAccessComponent implements OnInit, OnChanges {
   public exclusionsNotNamed: boolean = false;
   public legalOpsRoles: CaseRole[] = [];
   public namedLegalRoles: CaseRole[];
+  public namedAdminRoles: CaseRole[];
+  public adminRoles: CaseRole[];
```
Comment (scott-turner-hmcts) [inline src/cases/components/roles-and-access/roles-and-access.component.ts:17]: For consistency `adminRoles` and `namedAdminRoles` could have values set like with `legalOpsRoles` and `judicialRoles` properties e.g:

`public namedAdminRoles: CaseRole[] = [];`
`public adminRoles: CaseRole[] = [];`

File: src/role-access/containers/allocate-role/choose-role-error/choose-role-error.component.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+@Component({
+  selector: 'exui-choose-role-error',
+  templateUrl: './choose-role-error.component.html',
+})
+export class ChooseRoleErrorComponent implements OnInit {
+  @Input() public navEvent: AllocateRoleNavigation;
+
+  public roleCategory: string = '';
+  public jurisdiction: string;
```
Comment (scott-turner-hmcts) [inline src/role-access/containers/allocate-role/choose-role-error/choose-role-error.component.ts:17]: for consistency change similar to line above:

`public jurisdiction: string = ''`

or maybe change both lines as type is implicitly assigned from value:

`public roleCategory = '';`
`public jurisdiction = '';`

**Review by scott-turner-hmcts — CHANGES_REQUESTED** [review]
Added some minor comments to address

File: api/roleAccess/roleData.mock.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
+import { RoleCategory } from "./models/allocate-role.enum";
+
+export const mockAdminRoles: any[] = [{
```
[the remark this answers is not in the document]
Comment (connorpgpmcelroy) [reply api/roleAccess/roleData.mock.ts:3]: As both are mock data that directly corresponds to interfaces already defined in the angular layer, this is not necessary

File: src/cases/components/allocate-a-role/allocate-a-role-link.component.ts [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+@Component({
+    selector: 'exui-allocate-a-role-link',
+    templateUrl: './allocate-a-role-link.component.html',
+    styleUrls: ['./allocate-a-role-link.component.scss']
+  })
+
+export class AllocateARoleLinkComponent {
+    @Input() public routerLink: string;
+    @Input() public queryParams: string;
+    @Input() public showAllocateRoleLink: boolean = false;
```
[the remark this answers is not in the document]
Comment (connorpgpmcelroy) [reply src/cases/components/allocate-a-role/allocate-a-role-link.component.ts:13]: No, I'll replace

File: src/cases/components/role-access-section/role-access-section.component.html (hunk 1 above)
Comment (connorpgpmcelroy) [reply src/cases/components/role-access-section/role-access-section.component.html:10]: I'll investigate this - previously had problems with template not registering changes before but if suggested changes work I'll replace

File: src/cases/components/role-access-section/role-access-section.component.ts [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+@Component({
+    selector: 'exui-role-access-section',
+    templateUrl: './role-access-section.component.html'
+  })
+export class RoleAccessSectionComponent {
+    @Input() public title: string;
+    @Input() namedRoles: CaseRole [];
+    @Input() roles: CaseRole [];
+    @Input() public caseDetails: CaseView;
+    @Input() public showAllocateRoleLink: boolean = false;
```
[the remark this answers is not in the document]
Comment (connorpgpmcelroy) [reply src/cases/components/role-access-section/role-access-section.component.ts:14]: Will change

File: src/cases/components/roles-and-access/roles-and-access.component.ts (hunk 3 above)
Comment (connorpgpmcelroy) [reply src/cases/components/roles-and-access/roles-and-access.component.ts:17]: Sure

File: src/role-access/containers/allocate-role/choose-role-error/choose-role-error.component.ts (hunk 4 above)
Comment (connorpgpmcelroy) [reply src/role-access/containers/allocate-role/choose-role-error/choose-role-error.component.ts:17]: Will do

File: src/cases/components/role-access-section/role-access-section.component.html (hunk 1 above)
Comment (connorpgpmcelroy) [reply src/cases/components/role-access-section/role-access-section.component.html:10]: Have replaced this, although not directly on this component but the parent component and seems to be working

**Pull request merged**